src/morris_sensitivity.jl

```diff
@@ -1,15 +1,23 @@
-function generate_design_matrix(prob::DEProblem,p_range,p_steps,k = 10)
+struct Matrix
+    mat::Array{Array{Float64}}
+    spread::Float64
+end
+
+function generate_design_matrix(prob::DEProblem,p_range,p_steps;k = 10)
     ps = [linspace(p_range[i][1],p_range[i][2],p_steps[i]) for i in 1:length(p_range)]
     indices = [rand(1:i) for i in p_steps]
     all_idxs = Vector{typeof(indices)}(k)
-    all_idxs[1] = copy(indices)
-    i = 2
+    i = 1
     while i <= k
```

Comment:
If the index reflecting is done right, this should do one step each time, so it can be a for loop